lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2379,11 +2379,30 @@ namespace {
         hasMemberwiseInitializer = false;
       }
 
-      if (hasZeroInitializableStorage && !cxxRecordDecl) {
+      if (hasZeroInitializableStorage &&
+          !(cxxRecordDecl && cxxRecordDecl->hasDefaultConstructor())) {
```

Comment:
I wonder if we should limit this to C-looking C++ structs, e.g. those without base classes / templated parameters / etc.